components/raftstore/src/store/peer.rs

```diff
@@ -229,6 +230,7 @@ bitflags! {
         const SPLIT          = 0b0000_0010;
         const PREPARE_MERGE  = 0b0000_0100;
         const COMMIT_MERGE   = 0b0000_1000;
+        const NO_ADMIN       = 0b0001_0000;
```

Comment:
`split`, `prepare_merge` and `commit_merge` are exclusive. How about define the struct as a struct instead of bitflags? So we can avoid running out of bits.

Reply:
Do you mean to define the struct in protobuf? I think no need. We can just extend the bits type to u32 when it's exhausted and it doesn't break compatibility.

Reply:
Just in Rust:
```
pub struct ProposalContext {
    admin_type: AdminType,
    sync_log: bool,
}

pub enum AdminType {
    Split = 0b0000_0010,
    PREPARE_MERGE = 0b0000_0100,
    COMMIT_MERGE = 0b0000_1000,
}
```

Reply:
What's the difference? We still need to encode it as bitflags to keep compatibility.

Reply:
It needs to be encoded as a u8 for compatibility. But we can now assign 0b0000_0011 to new value.